packages/protocol/contracts/L1/libs/LibVerifying.sol

```diff
@@ -272,8 +159,29 @@ library LibVerifying {
                     resolver.resolve("tko_token", false)
                 );
 
-                _rewardProvers(config, fc, reward, taikoToken);
-                _refundProposerDeposit(target, tRelBp, taikoToken);
+                // reward the prover
+                if (reward != 0) {
+                    uint256 _reward = reward;
+                    if (taikoToken.balanceOf(fc.prover) == 0) {
+                        // Reduce reward to 1 wei as a penalty if the prover
+                        // has 0 TKO balance. This allows the next prover reward
+                        // to be fully paid.
+                        _reward = uint256(1);
+                    }
+                    taikoToken.mint(fc.prover, _reward);
```

Comment:
May as well do only rewards if `balance > 0`, because not the max gas consumption is still much higher in the worst case.

Perhaps we shouldn't even do the transfers as part of the block verification if an ERC20 is used because quite expensive! (I don't know if it would be worthwhile to make it a pull API where anyone could claim the rewards themselves, depends I guess how much state is need to make this possible, if ETH is used it is probably cheaper to just transfer here with are small gas limit).

Reply:
Changed as suggested